modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesRecoveryTask.java

```diff
@@ -23,129 +23,197 @@
-            }
-        }
-
-        fromMap.removeAll(fromDBDuplicated);
-        for (AndesExchange exchange : fromMap) {
-            for (ExchangeListener listener : exchangeListeners) {
-                log.warn("Recovering node. Removing exchange " + exchange.toString());
-                listener.handleClusterExchangesChanged(exchange, ExchangeListener.ExchangeChange.Deleted);
-            }
-        }
-    }
-
-    private void reloadQueuesFromDB() throws AndesException {
-        List<AndesQueue> fromDB = andesContextStore.getAllQueuesStored();
-        List<AndesQueue> fromMap = amqpConstructStore.getQueues();
-        List<AndesQueue> fromDBDuplicated = new ArrayList<AndesQueue>(fromDB);
-
-        fromDB.removeAll(fromMap);
-        for (AndesQueue queue : fromDB) {
-            for (QueueListener listener : queueListeners) {
-                log.warn("Recovering node. Adding queue " + queue.toString());
-                listener.handleClusterQueuesChanged(queue, QueueListener.QueueEvent.ADDED);
-            }
-        }
-
-        fromMap.removeAll(fromDBDuplicated);
-        for (AndesQueue queue : fromMap) {
-            for (QueueListener listener : queueListeners) {
-                log.warn("Recovering node. Removing queue " + queue.toString());
-                listener.handleClusterQueuesChanged(queue, QueueListener.QueueEvent.DELETED);
-            }
-        }
-    }
-
-    private void reloadBindingsFromDB() throws AndesException {
-        List<AndesExchange> exchanges = andesContextStore.getAllExchangesStored();
-        for (AndesExchange exchange : exchanges) {
-            List<AndesBinding> fromDB = andesContextStore.getBindingsStoredForExchange(exchange.exchangeName);
-            List<AndesBinding> fromMap = amqpConstructStore.getBindingsForExchange(exchange.exchangeName);
-            List<AndesBinding> fromDBDuplicated = new ArrayList<AndesBinding>(fromDB);
-            fromDB.removeAll(fromMap);
-            for (AndesBinding binding : fromDB) {
-                for (BindingListener listener : bindingListeners) {
-                    log.warn("Recovering node. Adding binding " + binding.toString());
-                    listener.handleClusterBindingsChanged(binding, BindingListener.BindingEvent.ADDED);
-                }
-            }
-
-            fromMap.removeAll(fromDBDuplicated);
-            for (AndesBinding binding : fromMap) {
-                for (BindingListener listener : bindingListeners) {
-                    log.warn("Recovering node. removing binding " + binding.toString());
-                    listener.handleClusterBindingsChanged(binding, BindingListener.BindingEvent.DELETED);
-                }
-            }
-        }
-    }
-
-    private void reloadSubscriptions() throws AndesException {
-        ClusterResourceHolder.getInstance().getSubscriptionManager().reloadSubscriptionsFromStorage();
-    }
+public class AndesRecoveryTask implements Runnable, StoreHealthListener {
```

Comment:
Checked the code. Tab character was used for indentation. Because of that, you will see a bit longer space in github.